.github/workflows/deploy-happy-stack.yml

```diff
@@ -95,6 +95,41 @@ jobs:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }}
         if: failure() && github.ref == 'refs/heads/main'
 
+  github-tag-and-release:
+    runs-on: ubuntu-22.04
+    needs:
+      - upgrade
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.deployment.sha }}
+      - name: Push Tag with latest prod commit SHA
+        uses: rickstaa/action-create-tag@v1
+        id: "tag_create"
+        with:
+          tag: ${{ github.event.deployment.payload.tag }}
+          tag_exists_error: false
+      - name: Build Changelog
+        id: build_changelog
+        uses: mikepenz/release-changelog-builder-action@v4
+        with:
+          configurationJson: |
+            {
+              "tag_resolver": {
+                "method": "sort",
+                "filter": {
+                  "pattern": "sha-(.+)",
+                  "flags": "gu"
+                }
+              }
+            }
+      - name: Produce Github Release for Prod Deployments
+        uses: softprops/action-gh-release@v1
+        if: steps.tag_create.outputs.tag_exists == 'false' && github.event.deployment.environment == 'prod'
```

Comment:
Is it intended that we run ` github-tag-and-release` for all deployments? 

`if: github.event.deployment.environment == 'prod'` could be moved to the [job level](https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution#overview) so we only run this for the prod environment.

Reply:
good catch, this was a relic from initially putting this step within the deployment step. Will fix